### ISSUE TITLE
Download hangs on linux (succeeds on windows)

### ISSUE THREAD

**Issue by craigmillernz** [SDFIdk/CDSETool#48]
Many thanks for providing this tool!
I am having the following problem..

Update, this appears to be similar to the problem described in this issue, but perhaps it wasn't tested on linux?
https://github.com/SDFIdk/CDSETool/issues/22

## Describe the bug
Download hangs when running in a container on linux, however succeeds when running on windows.

It appears the file successfuly downloads however an exception is thrown and the download does not close properly and the file is not present in the download path location.

See traceback below
```
[[  | 1 files in progress | 0 files done | 110.10 MB total downloaded | 9.44 MB/s ]]
S5P_NRTI_L2__SO2____20231206T012817_20231206T013317_31846_03_020601_20231206T022755.nc 117.38 MB (93%)                                                                        
[█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████           ]
Traceback (most recent call last):
  File "/usr/local/bin/tropomi_run.py", line 90, in <module>
    list(
  File "/opt/conda/envs/tropomi/lib/python3.10/site-packages/cdsetool/download.py", line 74, in download_features
    for feature in _concurrent_process(
  File "/opt/conda/envs/tropomi/lib/python3.10/site-packages/cdsetool/_processing.py", line 34, in _concurrent_process
    yield future.result()
  File "/opt/conda/envs/tropomi/lib/python3.10/concurrent/futures/_base.py", line 451, in result
    return self.__get_result()
  File "/opt/conda/envs/tropomi/lib/python3.10/concurrent/futures/_base.py", line 403, in __get_result
    raise self._exception
  File "/opt/conda/envs/tropomi/lib/python3.10/concurrent/futures/thread.py", line 58, in run

S5P_NRTI_L2__SO2____20231206T012817_20231206T013317_31846_03_020601_20231206T022755.nc (117.38 MB)
[[  | 0 files in progress | 1 files done | 117.38 MB total downloaded | 0.0 B/s ]]
^CException ignored in: <module 'threading' from '/opt/conda/envs/tropomi/lib/python3.10/threading.py'>
Traceback (most recent call last):
  File "/opt/conda/envs/tropomi/lib/python3.10/threading.py", line 1567, in _shutdown
    lock.acquire()
KeyboardInterrupt: 
```

### Steps to reproduce
Following the example as per the readme
```
features = query_features(
    "Sentinel5P",
    {
        "startDate": "2023-12-06",
        "completionDate": "2023-12-07",
        "processingLevel": "L2",
        "productType": "L2__SO2___",
        "geometry": shape_to_wkt("./NorthIsland.shp"),
    },
)

list(
        download_features(
            features,
            '/data',
            {   
                "concurrency": 4,
                "monitor": StatusMonitor(),
                "credentials": Credentials(CDSE_USER, CDSE_PASSWORD),
            },
        )
    )
```

### Expected behavior
I expected the download to terminate cleanly and not hang. 

### Environment
 - OS: [Linux Ubuntu 20.04]
 - Package version: [cdsetool==0.2.1]
 - Python version: [3.10.13]
 - Other details that you think may affect.

**Comment by pedep** [comment]
The machine i have been developing on is running Debian 12, so the problem is likely not specifically linux related.
My guess would be something specific to docker.

I ran a test in docker with the `"monitor": StatusMonitor(),` line commented out, and found this to be the full error message.

```
Traceback (most recent call last):
  File "/srv/test.py", line 18, in <module>
    list(
  File "/usr/local/lib/python3.12/site-packages/cdsetool/download.py", line 74, in download_features
    for feature in _concurrent_process(
  File "/usr/local/lib/python3.12/site-packages/cdsetool/_processing.py", line 34, in _concurrent_process
    yield future.result()
          ^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/concurrent/futures/_base.py", line 449, in result
    return self.__get_result()
           ^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/concurrent/futures/_base.py", line 401, in __get_result
    raise self._exception
  File "/usr/local/lib/python3.12/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/cdsetool/download.py", line 72, in _download_feature
    return download_feature(feature, path, options)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/cdsetool/download.py", line 53, in download_feature
    os.rename(tmp, os.path.join(path, filename.replace(".SAFE", ".zip")))
OSError: [Errno 18] Invalid cross-device link: '/tmp/tmpd076fdkz' -> 'output/S5P_NRTI_L2__SO2____20231206T102817_20231206T103317_31851_03_020601_20231206T122842.nc'
```

It seems that `os.rename` does not support cross-device moving. Since docker uses an overlayfs mount by default as the root filesystem inside containers, and the data directory is mounted as a volume, this would make the sense.

I have fixed the above issue in #50 and released it in `cdsetool==0.2.2`. Could i get you to verify if this has fixed the issues you experienced?

**Comment by craigmillernz** [comment]
Thank you for the very prompt attention to this.  This fixes the problem and the download now terminates as expected.
